MTA-7133-LLM-Proxy-Disconnection-Network-Isolation-Procedures (#381)
migtools/mta-documentation: docs/topics/developer-lightspeed/ref_llm-proxy-developer-impact.adoc

// Module included in the following assemblies:
//
// * docs/developer-lightspeed-guide/master.adoc

:_mod-docs-content-type: REFERENCE

[id="llm-proxy-developer-impact_{context}"]
= Expected behavior on developer workstations after proxy shutdown

[role="_abstract"]
After the Operator terminates the large language model (LLM) proxy pod, developers experience specific connection errors and behavior changes in their integrated development environment (IDE).

> Developers experience the following behaviors after the proxy is shut down:

Reviewer: ```suggestion
Review the expected connection errors and behavior changes in the developer integrated development environment (IDE) to understand the impact of proxy shutdown.

A proxy shutdown causes the following behavior changes:
```

> * Any in-progress LLM request from an IDE client receives an immediate `503 Service Unavailable` or `connection refused` error.

Reviewer: ```suggestion
* In-progress large language model (LLM) requests from an IDE client receive an immediate `503 Service Unavailable` or `connection refused` error.
```

* Developers cannot send further LLM requests until the proxy is re-enabled.
* Standard IDE functionality, such as code editing, file navigation, and build tools, is not affected. Only the Developer Lightspeed AI assistant stops responding.

> * Developers see an error indicator in the Lightspeed panel of their IDE. The exact message depends on the IDE client, but it is typically the following:

Reviewer: ```suggestion
* An error indicator appears in the Lightspeed panel of the IDE.
+
The exact message depends on the IDE client, but it typically resembles the following:
```
